Helm: Update sizing plans (#3035)
grafana/mimir: operations/helm/charts/mimir-distributed/CHANGELOG.md

## main / unreleased

* [CHANGE] **breaking change** Reduce the number of ingesters in small.yaml form 4 to 3. This should be more accurate size for the scale of 1M AS. Before upgrading refer to [Scaling down ingesters](https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters) to scale down `ingester-3`. Alternatively override the number of ingesters to 4. #3035
* [ENHANCEMENT] Metamonitoring: If enabled and no URL is configured, then metamonitoring metrics will be sent to
  Mimir under the `metamonitoring` tenant; this enhancement does not apply to GEM. #3176
* [ENHANCEMENT] Improve default rollout strategies. Now distributor, overrides_exporter, querier, query_frontend, admin_api, gateway, and graphite components can be upgraded more quickly and also can be rolled out with a single replica without downtime. #3029
* [ENHANCEMENT] Metamonitoring: make scrape interval configurable. #2945

> * [ENHANCEMENT] Update sizing plans (small.yaml, large.yaml, capped-small.yaml, capped-large.yaml). These reflect better how we recommend running Mimir and GEM in production. #3035

Reviewer: I think this should be  a `[CHANGE]` and add a disclaimer that some replica counts and resource requests have been reduced!

Author: they have all changed. Should I list all the changes? I'd rather just link to the diff from this PR. Do you think that's ok @krajorama ?

Reviewer: Yeah, I think linking to the PR is ok, also ppl can use helm diff...

Reviewer: Mention that some replica counts and resource requests and limits are reduced!

Author: done in https://github.com/grafana/mimir/pull/3035/commits/c064706717d32c7465b1fc6093242e3fff6dfebf

* [CHANGE] Nginx: replace topology key previously used in `podAntiAffinity` (`failure-domain.beta.kubernetes.io/zone`) with a different one `topologySpreadConstraints` (`kubernetes.io/hostname`). #2722
* [CHANGE] Use `topologySpreadConstraints` instead of `podAntiAffinity` by default. #2722
* [CHANGE] Ingresses for the GEM gateway and nginx will no longer render on Kubernetes versions <1.19. #2872

> * [CHANGE] **breaking change** Reduce the number of ingesters in small.yaml and capped-small.yaml form 4 to 3. This should be more accurate size for the scale of 1M AS. Before upgrading refer to [Scaling down ingesters](https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters) to scale down `ingester-3`. Alternatively override the number of ingesters to 4. #3035

Reviewer: ```suggestion
* [CHANGE] **breaking change** Reduce the number of ingesters in small.yaml and capped-small.yaml from 4 to 3. This should be a more accurate size for the scale of 1M AS. Before upgrading refer to [Scaling down ingesters](https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters) to scale down `ingester-3`. Alternatively override the number of ingesters to 4. #3035
```

                topologyKey: 'kubernetes.io/hostname'
     ```
* [CHANGE] Ingresses for the GEM gateway and nginx will no longer render on Kubernetes versions <1.19. #2872

> * [CHANGE] **breaking change** Reduce the number of ingesters in small.yaml and capped-small.yaml from 4 to 3. This should be a more accurate size for the scale of 1M AS. Before upgrading refer to [Scaling down ingesters](https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters) to scale down `ingester-3`. Alternatively override the number of ingesters to 4. #3035

Reviewer: ```suggestion
```

                topologyKey: 'kubernetes.io/hostname'
     ```
* [CHANGE] Ingresses for the GEM gateway and nginx will no longer render on Kubernetes versions <1.19. #2872

> * [CHANGE] **breaking change** Reduce the number of ingesters in small.yaml and capped-small.yaml from 4 to 3. This should be a more accurate size for the scale of 1M AS. Before upgrading refer to [Scaling down ingesters](https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters) to scale down `ingester-3`. Alternatively override the number of ingesters to 4. #3035

Reviewer: ```suggestion
```
